includes/class-sensei-temporary-user.php

```diff
@@ -120,6 +120,59 @@ public static function filter_learners_query( string $query ) {
 		);
 	}
 
+	/**
+	 * Detect if the e-mail attributes relate to an e-mail from a temporary user.
+	 *
+	 * @access private
+	 * @since  $$next-version$$
+	 *
+	 * @param array  $atts   Email attributes.
+	 * @param string $email_domain The email domain to search for.
+	 * @return boolean Whether to block the email or not.
+	 */
+	public static function should_block_email( $atts, $email_domain ) {
+		$emails = $atts['to'];
+		if ( ! is_array( $emails ) ) {
+			$emails = explode( ',', $emails );
+		}
+		if ( ! empty( $atts['headers'] ) ) {
+			$headers = $atts['headers'];
+			if ( ! is_array( $headers ) ) {
+				// Explode the headers out, so this function can take
+				// both string headers and an array of headers.
+				$temp_headers = explode( "\n", str_replace( "\r\n", "\n", $headers ) );
+			} else {
+				$temp_headers = $headers;
+			}
+			// If it's actually got contents.
+			if ( ! empty( $temp_headers ) ) {
+				foreach ( $temp_headers as $name => $content ) {
+					if ( is_int( $name ) && str_contains( $content, ':' ) ) {
+						list ( $name, $content) = explode( ':', trim( $content ), 2 );
+					}
+
+					// Cleanup crew.
+					$name    = trim( $name );
+					$content = trim( $content );
+
+					if ( in_array( strtolower( $name ), [ 'from', 'cc', 'bcc', 'reply-to' ], true ) ) {
+						$emails = array_merge( (array) $emails, explode( ',', $content ) );
+					}
+				}
+			}
+		}
+		foreach ( $emails as $address ) {
+			if ( preg_match( '/(.*)<(.+)>/', $address, $matches ) && count( $matches ) === 3 ) {
+				$address = $matches[2];
+			}
+			if ( str_ends_with( $address, '@' . $email_domain ) ) {
```

Comment:
Haha, the `str_ends_with` function was introduced in PHP 8.0 but it looks like the `symfony/polyfill-php80` is working. Nice!

Reply:
@m1ro Actually, even without `symfony/polyfill-php80` this would work well.

The reason? WordPress also implements some polyfills for functions from PHP 8.0...since WP 5.9. Here's the polyfill for `str_ends_with`: https://github.com/WordPress/WordPress/blob/f2d315036b0d4bba651852b4d32ffdd58e97b2f1/wp-includes/compat.php#L467-L489 :smile:

Reply:
Oh nice, TIL!